From aea858d4e8c99f5785dcac719c53b2fafd6e74bf Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Sat, 28 May 2022 20:41:05 +0200
Subject: [PATCH 4/6] counter.cpp: Move away from deprecated qsrand()

qsrand() has been deprecated as per https://doc.qt.io/qt-5/qtglobal-obsolete.html#qsrand

QRandomGenerator was introduced in Qt 5.10, so for older versions of Qt such as 5.6 used in SFOS keep using qsrand.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 libconnman-qt/counter.cpp | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/libconnman-qt/counter.cpp b/libconnman-qt/counter.cpp
index 4ebde7a..0b95ce6 100644
--- a/libconnman-qt/counter.cpp
+++ b/libconnman-qt/counter.cpp
@@ -8,6 +8,9 @@
  */
 
 #include <QtDBus/QDBusConnection>
+#if (QT_VERSION >= QT_VERSION_CHECK(5,10,0))
+#include <QRandomGenerator>
+#endif
 
 #include "counter.h"
 #include "networkmanager.h"
@@ -29,8 +32,12 @@ Counter::Counter(QObject *parent) :
     registered(false)
 {
     QTime time = QTime::currentTime();
-    qsrand((uint)time.msec());
-    int randomValue = qrand();
+    #if (QT_VERSION >= QT_VERSION_CHECK(5,10,0))
+        quint32 randomValue = QRandomGenerator::global()->generate();
+    #else
+        qsrand((uint)time.msec());
+        int randomValue = qrand();
+    #endif
     //this needs to be unique so we can use more than one at a time with different processes
     counterPath = "/ConnectivityCounter" + QString::number(randomValue);
 
-- 
2.36.1

